apps/prairielearn/elements/pl-file-preview/pl-file-preview.js

```diff
@@ -110,7 +110,24 @@
               const type = blob.type;
               if (type === 'text/plain') {
                 const text = await blob.text();
-                code.textContent = text;
+                if (escapedFileName.endsWith('.ipynb')) {
+                  // importing the notebookjs library doesn't return an object, it sets the global variable 'ns'
+                  // importing DOMPurify sets the global variable DOMPurify.
+                  await Promise.all([
+                    import('marked'),
```

Comment:
I have a bit of a reservation of introducing yet another markdown library. We're currently using showdown in pl-file-editor (markdown preview), pl-rich-text-editor and pl-xss-safe, and unified in server-side conversion. Can we settle on one of those?

Reply:
I get the following error when I try to dynamically load showdown:

`TypeError: undefined is not an object (evaluating 'this.showdown=x') — showdown.js:5153`

* Does this look familiar (i.e., might there be a quick, easy fix)?
* I could just statically load it; but, I don't think @nwalters512 would want this library loaded unless the file being previewed is .ipynb.

Reply:
To clarify: I don't want to block this use, particularly considering #6480, I'm just questioning if the use of a new library for functionality that is already in place. If there is no easy alternative please proceed.

Reply:
I understand. I agree that fewer libraries is better, it just a question of the best tradeoff: Fewer libraries, dynamic loading, or (potentially) waiting for a PR in showdown (assuming the error is their bug, not mine). I'm fine with whatever you prefer.

Reply:
@jonatanschroeder could we move in the other direction and switch everything else to use `marked`? I'm not sure what compatibility looks like, but it does support dynamic imports (good for lazy-loading when needed), and it's less than half the size ([10.6kB](https://bundlephobia.com/package/marked@14.0.0) vs [23.lkB](https://bundlephobia.com/package/showdown@2.1.0)).

This doesn't need to block this PR, of course.

Reply:
> @jonatanschroeder could we move in the other direction and switch everything else to use `marked`? I'm not sure what compatibility looks like, but it does support dynamic imports (good for lazy-loading when needed), and it's less than half the size ([10.6kB](https://bundlephobia.com/package/marked@14.0.0) vs [23.lkB](https://bundlephobia.com/package/showdown@2.1.0)).
> 
> This doesn't need to block this PR, of course.

Ideally I'd prefer to add bundling to @prairielearn/markdown, so we have consistency between server-side and client-side conversion, but marked may be an alternative if that package ends up being too big or if it takes too long to get this rolling. 

That said, one option we could also consider is to switch everything to marked, including server-side. Marked seems to have [the extensibility we need](https://marked.js.org/using_pro) to convert, e.g., pre to pl-code. It also has native support for [inline parsing](https://marked.js.org/using_advanced#inline), which is used in the context of rubrics, and that we currently handle in a somewhat hacky way. It doesn't handle sanitizing, but we can combine it with DOMPurify.

The only issue we might have with marked is that marked does not support HTML to markdown conversion, which is [used in pl-rich-text-editor element](https://github.com/PrairieLearn/PrairieLearn/blob/ee607c24e6461abe4abd4edb6a017292f9461da1/apps/prairielearn/elements/pl-rich-text-editor/pl-rich-text-editor.js#L123). This is a point where unified [could be used for this purpose](https://github.com/rehypejs/rehype-remark). That said, to be honest, with the problems we have had with showdown in this respect (see #5986, #10026), I would not be opposed to splitting this to use a different package altogether, potentially even in the Python side of pl-rich-text-editor itself (i.e., the parse function) instead of client-side.

Reply:
Added this discussion to #6480. And to be clear, please proceed with marked in this PR.